File: redis-jedis-demo/src/main/java/com/akatkar/java/learning/redis/demo/Example05_SortedSets.java
package com.akatkar.java.learning.redis.demo;

import java.util.HashMap;
import java.util.Map;
import redis.clients.jedis.Jedis;
import static com.akatkar.java.learning.redis.demo.Config.*;

/**
 * Sorted Sets are like a Set where each member has an associated ranking, that
 * is used for sorting them:
 */
public class Example05_SortedSets {

    public static void main(String[] args) {

        try(Jedis jedis = new Jedis(REDIS_HOST)) {

            Map<String, Double> scores = new HashMap<>();
            scores.put("PlayerOne", 3000.0);
            scores.put("PlayerTwo", 1500.0);
            scores.put("PlayerThree", 8200.0);

            scores.forEach((key, value) -> jedis.zadd("scores", value, key));

            String player = jedis.zrevrange("scores", 0, 1).iterator().next();
            long rank = jedis.zrevrank("scores", "PlayerOne");
            System.out.println("player:" + player + ", rank:" + rank);
        }
    }
}

File: python-redis-demo/examples/example05_sortedsets.py
import redis

cache = redis.StrictRedis(host='192.168.1.125', port=6379, db=0)

scores = dict()

scores["PlayerOne"] = 3000.0
scores["PlayerTwo"] = 2500.0
scores["PlayerThree"] = 7200.0

for player in scores:
    #print(player, scores.get(player))
    cache.zadd("scores", scores)

players = cache.zrevrange("scores", 0, 1)
print(players)

players = cache.zrange("scores", 0, -1, withscores=True)
print(players)

rank = cache.zrevrank("scores", "PlayerOne")
print(rank)